services/users/users.go

```diff
@@ -6,28 +6,51 @@ package users
 import (
 	"context"
 	"fmt"
+	"runtime"
 
+	"github.com/mattermost/mattermost-server/v5/einterfaces"
 	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/services/cache"
 	"github.com/mattermost/mattermost-server/v5/shared/i18n"
 	"github.com/mattermost/mattermost-server/v5/shared/mlog"
 	"github.com/mattermost/mattermost-server/v5/store"
 )
 
 type UserService struct {
-	store  store.UserStore
-	config func() *model.Config
+	store        store.UserStore
+	sessionStore store.SessionStore
+	sessionCache cache.Cache
+	metrics      einterfaces.MetricsInterface
+	cluster      einterfaces.ClusterInterface
+	config       func() *model.Config
 }
 
 type UserCreateOptions struct {
 	Guest      bool
 	FromImport bool
 }
 
-func New(s store.UserStore, cfgFn func() *model.Config) *UserService {
-	return &UserService{
-		store:  s,
-		config: cfgFn,
+func New(s store.UserStore, ss store.SessionStore, c einterfaces.ClusterInterface, m einterfaces.MetricsInterface, cfgFn func() *model.Config) (*UserService, error) {
+	cacheProvider := cache.NewProvider()
+	if err := cacheProvider.Connect(); err != nil {
+		return nil, fmt.Errorf("could not create cache provider: %w", err)
+	}
+
+	sessionCache, err := cacheProvider.NewCache(&cache.CacheOptions{
+		Size:           model.SESSION_CACHE_SIZE,
```

Comment:
Should `SESSION_CACHE_SIZE` be moved to this package?

Reply:
Checked and this is also used in `store` package [here](https://github.com/mattermost/mattermost-server/blob/46649292f8149dbfb4e29c55a3f40140717cc5b6/store/sqlstore/channel_store.go#L30)

Reply:
Hmm .. I'd say those things should really be unexported constants under the store package. But this is getting out of scope. Fine to leave it as-is.